packages/storage/amplify_storage_s3/lib/method_channel_storage_s3.dart

```diff
@@ -152,7 +152,12 @@ class AmplifyStorageS3MethodChannel extends AmplifyStorageS3 {
       if (onProgress != null) {
         _transferProgressionCallbackMap[request.uuid] = onProgress;
       }
-
+      // Delete the file if it already exists since iOS will only write to
+      // a non-existent file; it will not overwrite the contents of an existing
+      // one.
+      if (Platform.isIOS && await request.local.exists()) {
+        await request.local.delete();
+      }
```

Comment:
This is the actual fix. The rest is updating tests and Xcode files.

Reply:
If the download fails this will result in the original file being deleted without the new file being downloaded. Is that expected? Should this instead write to a temp directory and then copy it over?

Reply:
Since the intention is to overwrite whatever is passed in, then I think this is probably fine, although to be safe I can update it 👍